Lib/test/test_contextlib.py

```diff
@@ -1297,6 +1297,10 @@ def test_exception_groups(self):
                 [KeyError("ke1"), KeyError("ke2")],
             ),
         )
+        # Check handling of BaseExceptionGroup, using GeneratorExit so that
+        # we don't accidentally discard a ctrl-c with KeyboardInterrupt.
+        with suppress(GeneratorExit):
+            raise BaseExceptionGroup("message", [GeneratorExit()])
```

Comment:
The BaseExceptionGroup constructor can return an ExceptionGroup if all of the nested exceptions are Exception. So it would be nice to assert that the type is BaseException Group.

Can we have a couple of tests with a partial matches - swallow the BaseExceptions and let Exceptions through, and vice versa?